src/simcams/simulated.camera.c

```diff
@@ -157,6 +158,136 @@ im_fill_pattern(const struct ImageShape* const shape,
+        for (uint32_t x = 0; x < shape->dims.width; ++x) {
+            const size_t o = (size_t)shape->strides.width * x +
+                             (size_t)shape->strides.height * y;
+            const float dx = x - cx;
+            const float dx2 = dx * dx;
+            buf[o] =
+              (uint16_t)(127.0f *
+                         (sinf(6.28f * (t * 10.0f + (dx2 + dy2) * 1e-2f)) +
+                          1.0f));
+        }
+    }
+}
+
+static void
+im_fill_pattern_i16(const struct ImageShape* const shape,
+                    float ox,
+                    float oy,
+                    void* buf_)
+{
+    uint16_t* buf = (uint16_t*)buf_;
+    float t = get_animation_time_sec();
+
+    const float cx = ox + 0.5f * (float)shape->dims.width;
+    const float cy = oy + 0.5f * (float)shape->dims.height;
+    for (uint32_t y = 0; y < shape->dims.height; ++y) {
+        const float dy = y - cy;
+        const float dy2 = dy * dy;
+        for (uint32_t x = 0; x < shape->dims.width; ++x) {
+            const size_t o = (size_t)shape->strides.width * x +
+                             (size_t)shape->strides.height * y;
+            const float dx = x - cx;
+            const float dx2 = dx * dx;
+            buf[o] =
+              (int16_t)(127.0f *
+                        (sinf(6.28f * (t * 10.0f + (dx2 + dy2) * 1e-2f)) +
+                         1.0f));
+        }
+    }
+}
+
+static void
+im_fill_pattern_f32(const struct ImageShape* const shape,
+                    float ox,
+                    float oy,
+                    void* buf_)
+{
+    float* buf = (float*)buf_;
+    float t = get_animation_time_sec();
+
+    const float cx = ox + 0.5f * (float)shape->dims.width;
+    const float cy = oy + 0.5f * (float)shape->dims.height;
+    for (uint32_t y = 0; y < shape->dims.height; ++y) {
+        const float dy = y - cy;
+        const float dy2 = dy * dy;
+        for (uint32_t x = 0; x < shape->dims.width; ++x) {
+            const size_t o = (size_t)shape->strides.width * x +
+                             (size_t)shape->strides.height * y;
+            const float dx = x - cx;
+            const float dx2 = dx * dx;
+            buf[o] = (127.0f *
+                      (sinf(6.28f * (t * 10.0f + (dx2 + dy2) * 1e-2f)) + 1.0f));
+        }
+    }
+}
+
+static void
+im_fill_pattern(const struct ImageShape* const shape,
+                float ox,
+                float oy,
+                uint8_t* buf)
+{
+    switch (shape->type) {
+        case SampleType_i8:
+            im_fill_pattern_i8(shape, ox, oy, buf);
+            break;
+        case SampleType_u16:
+            im_fill_pattern_u16(shape, ox, oy, buf);
+            break;
+        case SampleType_i16:
+            im_fill_pattern_i16(shape, ox, oy, buf);
+            break;
+        case SampleType_f32:
+            im_fill_pattern_f32(shape, ox, oy, buf);
+            break;
+        default:
+            im_fill_pattern_u8(shape, ox, oy, buf);
```

Comment:
I don't like relying on `default` for this here. Better would be to have a case for SampleType_u8 and just do nothing for the default case.